src/static_array.cr

```diff
@@ -163,6 +163,25 @@ struct StaticArray(T, N)
     self
   end
 
+  def map
+    clone.map! { |e| yield e }
```

Comment:
Regardless of the benchmark, invoking `clone` is wrong because it invokes `clone` on each element in the static array. This is fine for numbers, but you don't really want to clone objects inside the array.

I'd do what @bcardiff suggests: create a new static array. These can use the `new(&block : Int32 -> T)` method (similar to how `Array#reverse` and `Array#map_with_index` are implemented)